### ISSUE TITLE
updateRoboVMXML strips 3rd party libraries from robovm.xml

### ISSUE THREAD

**Issue by obigu** [libgdx/libgdx#3063]
I'm using libGDX 1.5.6 and RoboVM 1.1.0 and Android Studio.

If you have 3rd party .a libraries (with frameworks it's ok) defined in the robovm.xml file such as:

```
<libs>
    <lib>build/libs/ios/libgdx.a</lib>
    <lib>build/libs/ios/libObjectAL.a</lib>
    <!-- Flurry Version 5.4.0 -->
    <lib>libs/libflurry.a</lib>
    <!-- AdMob -->
     <lib>libs/libadmob.a</lib>
  </libs>
```

They get stripped when building the project and then runtime execution fails because of missing dependencies. This is due to "build" task depending on "updateRoboVMXML" which doesn't keep the non libGDX native libraries and also, it strips all the comments from the file.

This is the output of running updateRoboVMXML task:
```
<libs>
    <lib>build/libs/ios/libgdx.a</lib>
    <lib>build/libs/ios/libObjectAL.a</lib>
  </libs>
```

As a workaround you can comment out the following line not to run the task automatically every time it builds.
```
// build.dependsOn updateRoboVMXML
```

The updateRoboVMXML is a non modified one as generated by the Setup Gradle UI.

```
// Updates a robovm.xml file.
task updateRoboVMXML << {
 def xml = file('robovm.xml')

 if (!xml.exists()) {
   return
 }

  // Find all native (*.a) libraries beneath libs
  def libtree = fileTree(dir: 'build/libs', include: '**/*.a')

  def config = new groovy.util.XmlParser().parse(xml)
  config.libs.each {libs ->
      libs.children().clear()
      libtree.each { File file ->
          libs.appendNode('lib', 'build/libs/ios/' + file.getName())
      }
  }

  def writer = new FileWriter(xml)
  def printer = new XmlNodePrinter(new PrintWriter(writer))
  printer.setPreserveWhitespace true
  printer.print(config)
}
```

It's hard to believe that this is the default behaviour as I'm sure other people are building iOS with Gradle and have 3rd party .a dependencies what makes me think there may be some misconfiguration on my side but my configuration is pretty standard.

**Comment by Tom-Ski** [comment]
This is the right behaviour, the LibGDX natives are the only ones that need to be manually added to the robovm.xml by us in that script.  If you are using third party libraries, you use bindings, and RoboVM knows how to deal with those, I beileve they merge the xmls from the bindings with the 'master' xml.  @badlogic will have to confirm that.

I guess you are not using a binding?  If not, create one?

**Comment by badlogic** [comment]
We should make libGDX use the bindings mechanism. i need to rewrite the
robovm backend build for that.

**Comment by obigu** [comment]
@Tom-Ski I was writing a longer answer but @badlogic has been faster :). I understand the issue applies and we can use the workaround in the meantime. Is that correct?

**Comment by badlogic** [comment]
yes, and we should open an issue for fixing the way we bundle our iOS
natives as well as how we defi e the frameworks we need.

**Comment by Tom-Ski** [comment]
Sounds fun.

**Comment by MobiDevelop** [comment]
@badlogic Was this resolved with the 1.6.1 release?

**Comment by Tom-Ski** [comment]
Aye
